pulsar-client-tools/src/main/java/org/apache/pulsar/shell/PulsarShell.java

```diff
@@ -414,18 +441,36 @@ public List<String> readCommand() {
                 return;
             }
             execState = ExecState.RUNNING;
-            final String line = words.stream().collect(Collectors.joining(" "));
-            if (StringUtils.isBlank(line)) {
+            final String inputLine = words.stream().collect(Collectors.joining(" "));
+            if (StringUtils.isBlank(inputLine)) {
                 continue;
             }
-            if (isQuitCommand(line)) {
+            if (isQuitCommand(inputLine)) {
+                if (shellMode != ShellMode.DEFAULT){
+                    output(String.format("Exiting from %s shell mode", shellMode.command), terminal);
+                    promptMessage = promptMessage.substring(0, promptMessage.lastIndexOf(shellMode.command) - 1);
```

Comment:
It might be better to keep the current shell mode out of the promptMessage and build the `prompt` from `promptMessage` and `shellMode.command` each time there are changes.

Reply:
yes, updated it.